feat: improve menu and seeds

YoelFerreyra opened the pull request:
# Description

This PR introduces improvements to the menu (sidebar and navigation), enhances the filtering experience in the inbox, and makes the database seed process more robust and idempotent.

# Main changes:
## Menu and Navigation:

Improved the DashboardSidebar component to support both mobile and desktop views, with a collapsible sidebar for desktop and a slide-out menu for mobile devices.
Adjusted icons for better clarity (e.g., replaced campaign icon, improved imports).
Cleaned up redundant code and enhanced responsiveness.
Inbox and Filtering:

Added a new filter option to display hidden messages in the inbox, with a toggle switch in the modal filter.
Updated filter logic to support the new "hidden" state for emails.
Cleaned up dropdown actions and removed unused imports/components.

## Seed Script:

Refactored the Prisma seed script to use an idempotent pattern (findOrCreate) to prevent duplicate records on repeated runs.
Added checks before inserting related data (campaigns, messages, accounts, etc.).
Improved logging for easier debugging.

## General Cleanup:

Removed unused imports and redundant code in several components.
Improved code readability and maintainability across multiple files.

Commit:
feat: improve menu and seeds

Commit:
fix: hidden messages

Commit:
feat: refactor email fetching and seeding logic for improved clarity and efficiency

Commit:
fix: correct destructuring of total and unread in getAllMessagesAction

Israel-Laguan commented on app/dashboard/inbox/actions/index.ts:
```diff
@@ -21,13 +22,13 @@ export const getAllMessagesAction = async (
     AND: [
       { deletedAt: null },
-      { hideAt: null },
+      ...(hidden ? [{ hideAt: { not: null } }] : [{ hideAt: null }]),
```
```suggestion
      { hideAt: hidden ? { not: null } : null }
```

YoelFerreyra replied:
Done

Israel-Laguan commented on prisma/seed.ts:
```diff
@@ -77,192 +101,222 @@ async function main() {
+    });
+  }
+
+  const campaign = await prisma.campaign.findFirst({ where: { companyId: company.id } });
```
This is not needed as company is above

YoelFerreyra replied:
Done

Commit:
fix: update hidden message filter logic for clarity and consistency
fix: simplify table height calculation in InboxDataTable component
fix: retrieve first existing campaign from the database
